File: C#.Net/JDI/Web/Selenium/Elements/Common/Button.cs
using System;
using Epam.JDI.Core;
using Epam.JDI.Core.Interfaces.Common;
using Epam.JDI.Web.Selenium.Base;
using Epam.JDI.Web.Selenium.Elements.Base;
using OpenQA.Selenium;

namespace Epam.JDI.Web.Selenium.Elements.Common
{
    public class Button : ClickableText, IButton
    {
        public Button() : this (null) { }
        public Button(By byLocator = null, IWebElement webElement = null)
            : base(byLocator, webElement) { }

        protected new Func<WebBaseElement, string> GetTextFunc =
           el => el.WebAvatar.FindImmediately(() => el.WebElement.GetAttribute("value"), "");
    }
}

File: jdi-uitest-gui/src/main/java/com/epam/jdi/uitests/gui/sikuli/elements/common/Button.java
package com.epam.jdi.uitests.gui.sikuli.elements.common;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.gui.sikuli.elements.base.ClickableText;
import org.sikuli.script.Pattern;

/**
 * Created by Natalia_Grebenshchikova on 1/18/2016.
 */
public class Button extends ClickableText implements IButton {
    public Button() {
    }

    public Button(Pattern pattern) {
        super(pattern);
    }

}
